hazelcast/src/main/java/com/hazelcast/concurrent/atomiclong/AtomicLongContainerCollector.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.concurrent.atomiclong;
+
+import com.hazelcast.config.AtomicLongConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.MergePolicyConfig;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.impl.merge.AbstractNonPartitionedContainerCollector;
+
+import java.util.Collection;
+import java.util.Map;
+
+import static com.hazelcast.util.MapUtil.createHashMap;
+
+class AtomicLongContainerCollector extends AbstractNonPartitionedContainerCollector<AtomicLongContainer> {
+
+    private final Config config;
+    private final Map<AtomicLongContainer, String> containerNames;
+    private final Map<AtomicLongContainer, MergePolicyConfig> containerPolicies;
+
+    AtomicLongContainerCollector(NodeEngine nodeEngine, Map<String, AtomicLongContainer> containers) {
+        super(nodeEngine, containers);
+        this.config = nodeEngine.getConfig();
+        this.containerNames = createHashMap(containers.size());
```

Comment:
These will have to be concurrent maps since they are accessed concurrently from different partition threads. Applies to other collectors which have them as well.